src/develop/develop.c

```diff
@@ -70,7 +70,7 @@ void dt_dev_init(dt_develop_t *dev, int32_t gui_attached)
 
   dt_image_init(&dev->image_storage);
   dev->image_status = dev->preview_status = dev->preview2_status = DT_DEV_PIXELPIPE_DIRTY;
-  dev->image_loading = dev->preview_loading = dev->preview2_loading = dev->history_updating = 0;
+  dev->image_loading = dev->preview_loading = dev->preview2_loading = dev->history_updating = dev->image_invalid = 0;
```

Comment:
I'm never happy with using int instead of boolean. At least when you see gboolean in the code you know it is a bi-state switch. Can you change all the int above to be boolean ? At least image_invalid which is clearly a boolean but I bet all others are also bi-sate variables.

Reply:
Ok, i'll take care of this. Although - image_invalid is probably the only one **not** being a boolean :-)